Update Key vault with security recommendation (#896)
Azure/ResourceModules: arm/Microsoft.KeyVault/vaults/readme.md

| `createMode` | string | `default` |  | Optional. The vault's create mode to indicate whether the vault need to be recovered or not. - recover or default. |
| `cuaId` | string |  |  | Optional. Customer Usage Attribution ID (GUID). This GUID must be previously registered |
| `diagnosticEventHubAuthorizationRuleId` | string |  |  | Optional. Resource ID of the diagnostic event hub authorization rule for the Event Hubs namespace in which the event hub should be created or streamed to. |

> | `diagnosticEventHubName` | string |  |  | Optional. Name of the diagnostic event hub within the namespace to which logs are streamed. Without this, an event hub is created for each log category. The security recommendation is to set diagnostic settings to send data to either storage account, log analytics workspace or event hub.  |

Reviewer: ```suggestion
| `diagnosticEventHubName` | string |  |  | Optional. Name of the diagnostic event hub within the namespace to which logs are streamed. Without this, an event hub is created for each log category. For security reasons, it is recommended to set diagnostic settings to send data to either storage account, log analytics workspace or event hub.  |
```

Reviewer: Update description in template too

| `diagnosticLogsRetentionInDays` | int | `365` |  | Optional. Specifies the number of days that logs will be kept for; a value of 0 will retain data indefinitely. |

> | `diagnosticStorageAccountId` | string |  |  | Optional. Resource ID of the diagnostic storage account. The security recommendation is to set diagnostic settings to send data to either storage account, log analytics workspace or event hub.  |

Reviewer: ```suggestion
| `diagnosticStorageAccountId` | string |  |  | Optional. Resource ID of the diagnostic storage account. For security reasons, it is recommended to set diagnostic settings to send data to either storage account, log analytics workspace or event hub.  |
```

Reviewer: Update description in template too

> | `diagnosticWorkspaceId` | string |  |  | Optional. Resource ID of the diagnostic log analytics workspace. The security recommendation is to set diagnostic settings to send data to either storage account, log analytics workspace or event hub.  |

Reviewer: ```suggestion
| `diagnosticWorkspaceId` | string |  |  | Optional. Resource ID of the diagnostic log analytics workspace. For security reasons, it is recommended to set diagnostic settings to send data to either storage account, log analytics workspace or event hub.  |
```

Reviewer: Update description in template too

| `logsToEnable` | array | `[AuditEvent]` | `[AuditEvent]` | Optional. The name of logs that will be streamed. |
| `metricsToEnable` | array | `[AllMetrics]` | `[AllMetrics]` | Optional. The name of metrics that will be streamed. |
| `name` | string |  |  | Optional. Name of the Key Vault. If no name is provided, then unique name will be created. |

> | `networkAcls` | object | `{object}` |  | Optional. Service endpoint object information. DefaultAction should set to Deny for security recommendation. |

Reviewer: ```suggestion
| `networkAcls` | object | `{object}` |  | Optional. Service endpoint object information. For security reasons, it is recommended to set the DefaultAction `Deny` |
```

Reviewer: Update description in template too

> | `privateEndpoints` | array | `[]` |  | Optional. Configuration Details for private endpoints. Security recommendation is to use private endpoints whenever possible.  |

Reviewer: ```suggestion
| `privateEndpoints` | array | `[]` |  | Optional. Configuration Details for private endpoints. For security reasons, it is reommended to use private endpoints whenever possible.  |
```

Reviewer: Update description in template too

## Outputs

> | Output Name             | Type   | Description                                                  |

Reviewer: Undo autoformatting. Seems you have a VSCode extension installed that does this for you. Please disable to autoformatting for that extension and roll the changes back
